### PR TITLE
New Job API

### PR DESCRIPTION
**Pull request by starcraftman**
Make us work with new api. To be merged after neovim merges PR neovim/neovim#2247.

Notes:
* Disabled python & ruby on nvim. The current implementations won't work. Not sure if there's any need, since job support in nvim is mandatory.
* Jettisoned the echo Error trick in favor of just checking return code on job exit.

### CODE REVIEW

**Comment by junegunn** [comment]
Great job! I'll merge when it's ready.

File: plug.vim [hunk 1]
```diff
@@ -863,18 +862,22 @@ function! s:spawn(name, cmd, opts)
...
-            \ (has_key(a:opts, 'dir') ? s:with_cd(a:cmd, a:opts.dir) : a:cmd)
-            \ . ' || echo Error'])
-    if x > 0
-      let job.jobid = x
-      augroup PlugJobControl
-        execute 'autocmd JobActivity' a:name printf('call s:job_handler(%s)', string(a:name))
-      augroup END
+    let opts = {
+        \ 'on_stdout': function('s:job_handler'),
+        \ 'on_exit' : function('s:job_handler'),
```
Comment (junegunn) [inline plug.vim]: Just a minor nitpicking. There's a space before colon.

Comment (starcraftman) [inline plug.vim]: Those sneaky spaces!

**Comment by tarruda** [comment]
@starcraftman there was a small change to the API("user" option was removed and the job options are passed as the "self" argument)

**Comment by starcraftman** [comment]
@tarruda Done. Seems like an improvement, easy to have user object simply contain the callbacks required.

**Comment by starcraftman** [comment]
@junegunn Ping.  You can pull now. Neovim PR went through, I don't see any drastic changes to API from my last update. Ran a test against neovim & Install/Update works fine.

**Comment by junegunn** [comment]
:tada:

**Pull request merged**